git-keeper.py

```diff
@@ -57,32 +66,34 @@ def git_clone_upload(
 ):
     logger.info("Processing repo: %s", repo_url)
     if not repo_url.endswith(".git"):
-        repo_url = repo_url + ".git"
+        repo_url += ".git"
         logger.debug("Appending .git to repo")
-    repo_dir = os.path.join(workdir, os.path.basename(repo_url))
-    repo_tar = repo_dir + ".tar"
+    repo_dir = Path(workdir) / Path(repo_url).name
+    repo_tar = str(repo_dir) + ".tar"
     logger.debug("Clearing workdir")
     cleanwrkdir(workdir)
     logger.debug("Clonning repo")
     clone_repo(repo_url, repo_dir)
     logger.debug("TARing repo")
     tar(repo_tar, repo_dir)
     repo_gpg = repo_tar + ".gpg"
-    with open(repo_tar, "rb") as f:
+    with Path.open(repo_tar, "rb") as f:
```

Comment:
why are you opening this with `Path` if `repo_tar` is a string? you may not want to make `repo_tar` a string in the first place...